blockchain/optimism.go

```diff
@@ -26,154 +10,169 @@ type OptimismClient struct {
-
-	opts.GasPrice = suggestedGasPrice
-
-	contractAddress, transaction, contractInstance, err := deployer(opts, o.Client)
-	if err != nil {
-		return nil, nil, nil, err
-	}
-
-	if err = o.ProcessTransaction(transaction); err != nil {
-		return nil, nil, nil, err
-	}
-
-	log.Info().
-		Str("Contract Address", contractAddress.Hex()).
-		Str("Contract Name", contractName).
-		Str("From", o.DefaultWallet.Address()).
-		Str("Total Gas Cost (OP)", utils.WeiToEther(transaction.Cost()).String()).
-		Str("Network Name", o.NetworkConfig.Name).
-		Msg("Deployed contract")
-	return &contractAddress, transaction, contractInstance, err
-}
-
-// Fund sends some ARB to an address using the default wallet
-func (o *OptimismClient) ReturnFunds(fromPrivateKey *ecdsa.PrivateKey) error {
-	to := common.HexToAddress(o.DefaultWallet.Address())
-
-	// Arbitrum uses legacy transactions and gas estimations
-	suggestedGasPrice, err := o.Client.SuggestGasPrice(context.Background())
-	if err != nil {
-		return err
-	}
-	fromAddress, err := utils.PrivateKeyToAddress(fromPrivateKey)
-	if err != nil {
-		return err
-	}
-
-	balance, err := o.Client.BalanceAt(context.Background(), fromAddress, nil)
-	if err != nil {
-		return err
-	}
-	estimatedGasCost := big.NewInt(1).Mul(suggestedGasPrice, big.NewInt(21000))
-	// Optimism needs to calculate both the L1 and L2 gas fees
-	// https://community.optimism.io/docs/developers/build/transaction-fees/#the-l1-data-fee
-	optimismL1GasContract := common.HexToAddress("0x420000000000000000000000000000000000000F")
-	optimismGasContract, err := contracts.NewOptimismGas(optimismL1GasContract, o.Client)
-	if err != nil {
-		return err
-	}
-	l1Fee, err := optimismGasContract.GetL1Fee(&bind.CallOpts{}, types.LegacyTx{}.Data)
-	if err != nil {
-		return err
-	}
-	// Optimism's L1 gas estimation has an error margin of 25%, we use 26 for rounding errors
-	// https://help.optimism.io/hc/en-us/articles/4416677738907
-	l1FeeFloat := big.NewFloat(1).SetUint64(l1Fee.Uint64())
-	l1FeeFloat.Mul(l1FeeFloat, big.NewFloat(1.26))
-	l1FeeFloatUint, _ := l1FeeFloat.Uint64()
-	l1Fee.SetUint64(l1FeeFloatUint)
-
-	estimatedGasCost.Add(estimatedGasCost, l1Fee)
-	balance.Sub(balance, estimatedGasCost)
-
-	nonce, err := o.GetNonce(context.Background(), fromAddress)
-	if err != nil {
-		return err
-	}
-
-	tx, err := types.SignNewTx(fromPrivateKey, types.LatestSignerForChainID(o.GetChainID()), &types.LegacyTx{
-		Nonce:    nonce,
-		To:       &to,
-		Value:    balance,
-		GasPrice: suggestedGasPrice,
-		Gas:      21000,
-	})
-	if err != nil {
-		return err
-	}
-
-	log.Info().
-		Str("Token", "OP").
-		Str("From", fromAddress.Hex()).
-		Uint64("Amount", balance.Uint64()).
-		Uint64("Estimated Gas Cost", estimatedGasCost.Uint64()).
-		Msg("Returning Funds to Default Wallet")
-	if err := o.Client.SendTransaction(context.Background(), tx); err != nil {
-		return err
-	}
-
-	return o.ProcessTransaction(tx)
-}
+// // Fund sends some OP to an address using the default wallet
```

Comment:
Do we need to keep this?

Reply:
Not for long. Optimism Bedrock recently updated which changed a lot of their gas calculation, but they haven't fully released all the modifications yet. So I'm sitting on this just in case I need to re-use some of it. Will remove it when I get further info from Optimism team.